lib/generator.js

```diff
@@ -975,6 +993,9 @@
                 clipToDocumentBounds: settings.clipToDocumentBounds || false,
                 maxDimension: settings.maxDimension || 10000
             };
+        if (settings.hasOwnProperty("thread")) {
```

Comment:
yeah, for the most part we want thread to be undefined so PS can pick wether to thread or not. I can comment above that line if you like